js/app.js

```diff
@@ -0,0 +1,108 @@
+/* Pedimos al usuario una frase para encriptarla */
+var sentence = prompt('Escriba una frase: ');
+/* Verifica si escribio algo */
+if (sentence === ' ') {
+  alert('Escribe una frase');
+}
+/* Creamos una funcion, para Cifrar una frase */
+function cipher(sentence, number) {
+  /* Verifica si escribio solo letras */
+  if (sentence === parseInt(sentence)) {
+    alert('Escribe solo letras');
+  } else {
+    /* Almacenamos los newCodeAsciiOfLetterUp y newCodeAsciiOfLetterDown */
+    var arrayOfCodeAsciiLetter = [];
+    /* Hacemos un bucle para extraer los caracteres de la sentence */
+    for (i = 0; i < sentence.length; i++) {
+      /* Extraemos una letter de la sentence de acuerdo a su posición */
+      var letter = sentence.charAt(i);
+      /* Verificamos si la letra es mayuscula o miniscula */
+      if (letter === letter.toUpperCase()) {
+        /* Devuelve el Codigo ASCII de la letra mayuscula */
+        var codeAsciiOfLetterUp = letter.charCodeAt(0);
+        /* Aplicando la formula nos devuelve el nuevo código ASCII */
+        newCodeAsciiOfLetterUp = (codeAsciiOfLetterUp - 65 + n) % 26 + 65;
```

Comment:
La programa no funciona porque hay error: 'ReferenceError: n is not defined'
La variable n nunca era declarado.

Reply:
Ah, el problema es que la parameter es llamada 'number', no 'n'